docs/backend-introduction.md

```diff
@@ -94,11 +94,11 @@ Important apps are for example:
 
 #### Stores
```

Comment:
That is actually completely out of date - pickups moved out of the stores module a long time ago, and stores are called places now. I wonder if we should remove a lot of this file?

Reply:
I reverted the whole backend introduction as this PR doesn't touch the backend.